raitools/raitools/raianalyzer/raianalyzer.py

```diff
@@ -135,12 +156,85 @@ def list(self):
+        """
+        # save each of the individual managers
+        for manager in self._managers:
+            manager.save(Path(path) / manager.name)
+        # save current state
+        self.train.to_pickle(Path(path) / TRAIN)
+        self.test.to_pickle(Path(path) / TEST)
+        meta = {TARGET_COLUMN: self.target_column,
+                TASK_TYPE: self.task_type}
+        with open(Path(path) / META_JSON, 'w') as file:
+            json.dump(meta, file)
+        if self._serializer is not None:
+            model_data = self._serializer.save(self.model)
+            # save the serializer
+            with open(Path(path) / SERIALIZER, 'wb') as file:
+                pickle.dump(self._serializer, file)
+            # save the model
+            with open(Path(path) / MODEL_PKL, 'w') as file:
+                file.write(model_data)
+        else:
+            has_setstate = hasattr(self.model, '__setstate__')
+            has_getstate = hasattr(self.model, '__getstate__')
+            if not (has_setstate and has_getstate):
+                raise ValueError(
+                    "Model must be picklable or a custom serializer must"
+                    " be specified")
+            with open(Path(path) / MODEL_PKL, 'wb') as file:
+                pickle.dump(self.model, file)
 
     @staticmethod
     def load(path):
-        # TODO: implement
-        raise NotImplementedError(
-            "Load not implemented for RAIAnalyzer")
+        """Load the RAIAnalyzer from the given path.
+
+        :param path: The path to load the RAIAnalyzer from.
+        :type path: str
+        """
+        # create the RAIAnalyzer without any properties using the __new__
+        # function, similar to pickle
+        inst = RAIAnalyzer.__new__(RAIAnalyzer)
+        top_dir = Path(path)
+        # load current state
+        train = pd.read_pickle(top_dir / TRAIN)
+        inst.__dict__[TRAIN] = train
+        inst.__dict__[TEST] = pd.read_pickle(top_dir / TEST)
+        with open(top_dir / META_JSON, 'r') as meta_file:
+            meta = meta_file.read()
+        meta = json.loads(meta)
+        target_column = meta[TARGET_COLUMN]
+        inst.__dict__[TARGET_COLUMN] = target_column
+        inst.__dict__[TASK_TYPE] = meta[TASK_TYPE]
+        inst.__dict__['_' + CLASSES] = train[target_column].unique()
+        serializer_path = top_dir / SERIALIZER
+        if serializer_path.exists():
+            with open(serializer_path) as file:
+                serializer = pickle.load(file)
+            inst.__dict__['_' + SERIALIZER] = serializer
+            with open(top_dir / MODEL_PKL, 'rb') as file:
+                inst.__dict__[MODEL] = serializer.load(file)
+        else:
+            inst.__dict__['_' + SERIALIZER] = None
+            with open(top_dir / MODEL_PKL, 'rb') as file:
+                inst.__dict__[MODEL] = pickle.load(file)
+        # load each of the individual managers
+        manager_names = ManagerNames.get_managers()
```

Comment:
Something feels strange about this. We have the list of names being extracted here, but then the `kvp_manager` has to list the managers explicitly.

Reply:
I need both manager names and the uninitialized manager classes.  I don't want to add a second name static property to managers, since I already have a name property on the managers (but I can't call it until they are initialized).  I thought getting the list from the enums might be a cleaner way to do it.

Reply:
I removed get_managers